src/synchronizeContacts.js

```diff
@@ -1,30 +1,202 @@
-const synchronizeContacts = (
-  localContacts,
-  remoteContacts,
-  synchronizationStrategy
+const get = require('lodash/get')
+const union = require('lodash/union')
+
+const { DOCTYPE_CONTACTS, APP_NAME } = require('./constants')
+const getInitialMetadata = require('./getInitialMetadata')
+const mergeContact = require('./mergeContact')
+const transpiler = require('./transpiler')
+
+const findGoogleContact = (cozyContact, googleContacts, contactAccountId) => {
+  const resourceName = get(
+    cozyContact,
+    ['cozyMetadata', 'sync', contactAccountId, 'id'],
+    null
+  )
+
+  if (resourceName) {
+    return googleContacts.find(
+      googleContact => googleContact.resourceName === resourceName
+    )
+  }
+
+  return undefined
+}
+
+const findCozyContact = (googleContact, cozyContacts, contactAccountId) => {
+  return cozyContacts.find(
+    contact =>
+      get(contact, ['cozyMetadata', 'sync', contactAccountId, 'id']) ===
+      googleContact.resourceName
+  )
+}
+
+const updateCozyMetadata = (
+  cozyContact,
+  etag,
+  resourceName,
+  contactAccountId
+) => {
+  const now = new Date().toISOString()
+  const updatedByApps = union(
+    get(cozyContact, 'cozyMetadata.updatedByApps', []),
+    [APP_NAME]
+  )
+
+  return {
+    ...cozyContact,
+    cozyMetadata: {
+      ...cozyContact.cozyMetadata,
+      updatedAt: now,
+      updatedByApps,
+      sync: {
+        [contactAccountId]: {
+          konnector: APP_NAME,
+          lastSync: now,
+          remoteRev: etag,
+          id: resourceName,
+          contactsAccountsId: contactAccountId
+        }
+      }
+    }
+  }
+}
+
+const shouldCreateOnGoogle = (cozyContact, contactAccountId) => {
+  return (
+    !cozyContact.cozyMetadata.sync ||
+    cozyContact.cozyMetadata.sync[contactAccountId] === undefined
+  )
+}
+
+const shouldCreateOnCozy = (cozyContact, googleContact) =>
+  !cozyContact && !googleContact.metadata.deleted
```

Comment:
Are we sure to always have `.metadata.deleted` ? Should we use `get` to be defensive ?

Reply:
I think we should have `metedata.deleted` but `get` is definitely a safer/better option!